### DISCUSSION TITLE
Multiple Timezone Interlaying Configuration

### DISCUSSION THREAD

**Discussion by yikun-chi** [carissalow/rapids#190] [General]
Hi, 

The datasets I am working with contain data with interlaying timezone. For example, a person can start in America/Chicago, travel to Europe/Berlin, and finally back to America/Chicago for a bit more time. Based on the documentation around timezone configuration, my tzcodes_files will have the following entry: 

- device_id_A, America/Chicago, 1587500000000
- device_id_A, Europe/Berlin,1587600000000
- device_id_A, America/Chicago,1587700000000

In this case, will RAPIDS treat this data as having three separate timezone intervals, or just two timezone intervals? Would this interlaying cause any problems for the pipeline? 

Thanks

**Comment by JulioV** [comment]
Hi, these are interpreted as three intervals. You can see each entry as the point at which the new timezone kicks in. In this case we would have:

- Chicago from `1587500000000` to `15876000000000-1`
- Berlin from `1587600000000` to `1587700000000-1`
- Chicago from `1587700000000` to current time

These intervals should not cause any issues with smartphone data. Fitbit data will have inaccurate features on the days the timezone changed because its data is saved in local time with no timezone.

**Reply by yikun-chi** [reply]
Thank you so much!
